sklearn/preprocessing/label.py

```diff
@@ -891,7 +895,9 @@ def transform(self, y):
         """
         check_is_fitted(self, 'classes_')
 
-        class_to_index = dict(zip(self.classes_, range(len(self.classes_))))
+        if self._indexed_classes_cache is None:
+            self._indexed_classes_cache = dict(zip(self.classes_, range(len(self.classes_))))
```

Comment:
The cache needs to be invalidated if fit is called again. Why not just do this caching in fit?

Reply:
Ah.. so basically we should take the cached value only in case if  `self.classes_` is the same, ie if we've already calculated that `dict(zip(` thing before?